### ISSUE TITLE
Yolo3 train: weighted specific class targeted

### ISSUE THREAD

**Issue by MBocchi** [dmlc/gluon-cv#1163]
Hi,
I have train **Yolo3** with custom dataset and custom classes (only **5 classes**). It works fine but now I would like to train my model adding a weight (in loss function by example), but only for one class.
I have seen that we can add a scalar global weight to the loss function but it is not what I am looking for. Do you  know how to do it?

Best regards

**Comment by ElectronicElephant** [comment]
You can refer to the target generators. AFAIK, there is a weight_t which you can modify.

**Comment by MBocchi** [comment]
Thank you @ElectronicElephant, indeed I've found in `YOLOV3PrefetchTargetGenerator` that it calculates `weights` attributes with the following definition:

> weights: element-wise gradient weights for center_targets and scale_targets.

The original formula for an annotation is: 

     weights[b, index, match, :] = 2.0 - gtw * gth / orig_width / orig_height

So if I add this for one specific class:

    weights[b, index, match, :] *= 1.5

and this for ther other classes:

    weights[b, index, match, :] *= 0.5

The training will more learn my wanted class than the others. Is it right?

**Comment by ElectronicElephant** [comment]
> 
> 
> Thank you @ElectronicElephant, indeed I've found in `YOLOV3PrefetchTargetGenerator` that it calculates `weights` attributes with the following definition:
> 
> > weights: element-wise gradient weights for center_targets and scale_targets.
> 
> The original formula for an annotation is:
> 
> ```
>  weights[b, index, match, :] = 2.0 - gtw * gth / orig_width / orig_height
> ```
> 
> So if I add for one specific class:
> 
> ```
> weights[b, index, match, :] *= 1.5
> ```
> 
> and for ther other classes:
> 
> ```
> weights[b, index, match, :] *= 0.5
> ```
> 
> The training will more learn my wanted class than the others. Is it right?

Yes, theoretically.
